pkg/apis/admissionregistration/types.go

```diff
@@ -123,8 +124,78 @@ type ValidatingAdmissionPolicy struct {
 	metav1.ObjectMeta
 	// Specification of the desired behavior of the ValidatingAdmissionPolicy.
 	Spec ValidatingAdmissionPolicySpec
+	// The status of the ValidatingAdmissionPolicy, including warnings that are useful to determine if the policy
+	// behaves in the expected way.
+	// Populated by the system.
+	// Read-only.
+	// +optional
+	Status ValidatingAdmissionPolicyStatus
+}
+
+// ValidatingAdmissionPolicyStatus represents the status of an admission validation policy.
+type ValidatingAdmissionPolicyStatus struct {
+	// The generation observed by the controller.
+	// +optional
+	ObservedGeneration int64
+	// The results of type checking for each expression.
+	// Presence of this field indicates the completion of the type checking.
+	// +optional
+	TypeChecking *TypeChecking
+	// The conditions represent the latest available observations of a policy's current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []ValidatingAdmissionPolicyCondition
+}
+
+// ValidatingAdmissionPolicyCondition describes the state of an admission validation policy at a certain point.
+type ValidatingAdmissionPolicyCondition struct {
+	// The type of deployment condition.
+	Type ValidatingAdmissionPolicyConditionType
+	// The status of the condition, one of True, False, Unknown.
+	Status api.ConditionStatus
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time
+	// The reason for the condition's last transition.
+	Reason string
+	// A human-readable message indicating details about the transition.
+	Message string
 }
 
+// ValidatingAdmissionPolicyConditionType is the condition type of admission validation policy.
+type ValidatingAdmissionPolicyConditionType string
+
+// TypeChecking contains results of type checking the expressions in the
+// ValidatingAdmissionPolicy
+type TypeChecking struct {
+	// The type checking warnings for each expression.
+	// +optional
+	// +listType=atomic
+	ExpressionWarnings []ExpressionWarning
+}
+
+// ExpressionWarning is a warning information that targets a specific expression.
+type ExpressionWarning struct {
+	// The expression that this warning is associated with.
+	Expression string
+	// The field that refers the expression.
+	RefField string
```

Comment:
The documentation doesn't make it clear what values to expect here, or how it is intended to be used... are these full field refs ("spec.validations[0].expression") or something else?

Reply:
I had a discussion with @cici37 . I am now switching to use json path in this field and remove the expression above.